File: Assets/Bundles/Shared/Resources/TS/Generator/GeneratorUtil.lua
-- Compiled with unity-ts v2.1.0-75
local GENERATOR_PICKUP_RANGE = require("Shared/TS/Generator/GeneratorMeta").GENERATOR_PICKUP_RANGE
-- * Set of utilities pertaining to generators.
local GeneratorUtil
do
	GeneratorUtil = setmetatable({}, {
		__tostring = function()
			return "GeneratorUtil"
		end,
	})
	GeneratorUtil.__index = GeneratorUtil
	function GeneratorUtil.new(...)
		local self = setmetatable({}, GeneratorUtil)
		return self:constructor(...) or self
	end
	function GeneratorUtil:constructor()
	end
	function GeneratorUtil:CanPickupGeneratorStack(generatorStateDto, entityPosition)
		local generatorPosition = generatorStateDto.pos
		local _entityPosition = entityPosition
		local distFromGenerator = (generatorPosition - _entityPosition).magnitude
		return distFromGenerator <= GENERATOR_PICKUP_RANGE
	end
end
return {
	GeneratorUtil = GeneratorUtil,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/Bundles/Imports/Core/Core~/src/Shared/Generator/GeneratorUtil.ts
import { GENERATOR_PICKUP_RANGE, GeneratorDto } from "./GeneratorMeta";

/** Set of utilities pertaining to generators. */
export class GeneratorUtil {
	/**
	 * Check if an entity position is in pick up range of generator stack.
	 * @param generatorStateDto A generator state.
	 * @param entityPosition Position of entity querying generator.
	 * @returns Whether or not entity is close enough to pick up generator stack.
	 */
	public static CanPickupGeneratorStack(generatorStateDto: GeneratorDto, entityPosition: Vector3): boolean {
		const generatorPosition = generatorStateDto.pos;
		const distFromGenerator = generatorPosition.sub(entityPosition).magnitude;
		return distFromGenerator <= GENERATOR_PICKUP_RANGE;
	}
}
